TomTomSDKExamples/Examples/UseCase/Offline/OfflineNavigationContent.swift

```diff
@@ -0,0 +1,689 @@
+            roadShieldReferences: .all,
+            announcementPoints: .all,
+            phoneticsType: .IPA
+        )
+
+        let options = try RoutePlanningOptions(
+            itinerary: itinerary,
+            costModel: costModel,
+            guidanceOptions: guidanceOptions
+        )
+        return options
+    }
+
+    private func planRoute(
+        from origin: CLLocationCoordinate2D,
+        to destination: CLLocationCoordinate2D
+    ) async throws
+        -> TomTomSDKNavigationEngines.RoutePlan
+    {
+        let routePlanningOptions = try createRoutePlanningOptions(from: origin, to: destination)
+        let route = try await planRoute(withRoutePlanner: routePlanner, routePlanningOptions: routePlanningOptions)
+        return TomTomSDKNavigationEngines.RoutePlan(route: route, routingOptions: routePlanningOptions)
+    }
+
+    private func planRoute(
+        withRoutePlanner routePlanner: OnlineRoutePlanner,
+        routePlanningOptions: RoutePlanningOptions
+    ) async throws
+        -> TomTomSDKRoute.Route
+    {
+        return try await withCheckedThrowingContinuation { (continuation: CheckedContinuation<TomTomSDKRoute.Route, Error>) in
+            routePlanner.planRoute(options: routePlanningOptions, onRouteReady: nil) { result in
+                switch result {
+                case let .failure(error):
+                    if let routingError = error as? RoutingError {
+                        print("Error code: \(routingError.code)")
+                        print("Error message: \(String(describing: routingError.errorDescription))")
+                        continuation.resume(throwing: routingError)
+                        return
+                    }
+                    continuation.resume(throwing: error)
+                case let .success(response):
+                    guard let routes = response.routes else {
+                        continuation.resume(throwing: RoutePlanError.unableToPlanRoute())
+                        return
+                    }
+                    guard let route = routes.first else {
+                        continuation.resume(throwing: RoutePlanError.unableToPlanRoute())
+                        return
+                    }
+                    continuation.resume(returning: route)
+                }
+            }
+        }
+    }
+}
+
+/// Update the OfflineMapCoordinator extension with the observe function to display changes to the current route and its progress on the map.
+///
+/// After adding a route on the map, you will receive a reference to that route.
+/// The addRouteToMap function keeps it in the routeOnMap so that it can show the visual progress along the route, without being added again.
+/// The default location provider has an approximate position but no route information.
+/// For smooth movement of the chevron along the route, ensure the map uses a mapMatchedLocationProvider from Navigation instead of the SimulatedLocationProvider.
```

Comment:
```suggestion
/// For smooth movement of the chevron along the route, ensure the map uses a `mapMatchedLocationProvider` from `Navigation` instead of the `SimulatedLocationProvider`.
```

Reply:
👍 Done!